Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -509,6 +528,58 @@ namespace AzToolsFramework
         PaintBrushSettingsRequestBus::Broadcast(&PaintBrushSettingsRequestBus::Events::SetColor, brushColor);
     }
 
+    AZStd::vector<float> PaintBrushManipulator::CalculateGaussianWeights(size_t smoothingRadius)
+    {
+        // The Gaussian function is G(x, y) = (1 / 2*pi*sigma^2) * e^(-(x^2 + y^2) / (2*sigma^2)).
+        // This function produces a bell curve symmetric around the origin, where (1 / 2*pi*sigma^2) is the height of the peak,
+        // and sigma (the standard deviation) controls the width of the peak.
+        // x and y are the relative distances from the center.
+
+        // This will contain all of the Gaussian weights that we compute and return.
+        AZStd::vector<float> gaussianWeights;
+
+        // We track the sum of all the weights so that we can normalize them at the end.
+        float sum = 0.0f;
+
+        // Keep a floating-point version of this so that we don't have to do int/float conversions in our loops below.
+        const float radius = aznumeric_cast<float>(smoothingRadius);
+
+        // In practice, only pixels within a distance of 3 * sigma affect the result, so we'll pick a sigma that's 1/3 of the distance
+        // in each direction.
+        const float sigma = AZStd::max(smoothingRadius / 3.0f, 1.0f);
```

Comment:
nit: better to use radius here to avoid the cast. Compiler probably optimizes, but still seems weird to not use the value you just saved off to avoid unnecessary casts.